### PR TITLE
remove null value in raw-metadatag

### PR DESCRIPTION
**Pull request by 928PJY**
#3251

### CODE REVIEW

File: src/docfx/lib/JsonUtility.cs [hunk 1]
```diff
@@ -184,6 +184,12 @@ public static JObject Merge(JObject first, IEnumerable<JObject> objs)
             return result;
         }
 
+        public static JObject ValidateNullValue(this JObject graph)
```
Comment (yufeih) [inline src/docfx/lib/JsonUtility.cs]: `JToken RemoveNulls(JToken)`

File: src/docfx/lib/JsonUtility.cs [hunk 2]
```diff
@@ -184,6 +184,12 @@ public static JObject Merge(JObject first, IEnumerable<JObject> objs)
...
+        {
+            var (_, jtoken) = ((JToken)graph).ValidateNullValue();
```
Comment (superyyrrzz) [inline src/docfx/lib/JsonUtility.cs]: >ValidateNullValue [](start = 46, length = 17)

We need to remove null silently in LegacyMarkdown. Will this method log error if finding null?

File: src/docfx/lib/JsonUtility.cs (hunk 1 above)
Comment (live1206) [reply src/docfx/lib/JsonUtility.cs]: +1

---
In reply to: [211813607](https://github.com/dotnet/docfx/pull/3252#discussion_r211813607) [](ancestors = 211813607)

File: src/docfx/lib/JsonUtility.cs (hunk 2 above)
Comment (OsmondJiang) [inline src/docfx/lib/JsonUtility.cs]: >ValidateNullValue [](start = 46, length = 17)

why ’validate‘  changed the input? chaneg to validateAndRemoveNullValue()?

File: test/docfx.Test/lib/JsonUtilityTest.cs [hunk 3]
```diff
@@ -104,6 +104,15 @@ public void TestListOfBasicClass()
             }
         }
 
+        [Fact]
+        public void TestValidateNullValue()
```
Comment (yufeih) [inline test/docfx.Test/lib/JsonUtilityTest.cs]: This should have been tested, remove?

**Pull request merged**